plugins/org.jkiss.dbeaver.ext.mysql.ui/src/org/jkiss/dbeaver/ext/mysql/ui/controls/PrivilegeTableControl.java

```diff
@@ -236,10 +236,12 @@ public void fillGrants(List<MySQLGrant> grants) {
                 if (isStatic && !grant.isStatic()) {
                     continue;
                 }
-                if (privilege.getName().equalsIgnoreCase(MySQLConstants.PRIVILEGE_GRANT_OPTION_NAME) && grant.isGrantOption()) {
-                    // WITH GRANT OPTION is enabled only in this case
-                    privilegeEnabled = true;
-                    break;
+                if (privilege.getName().equalsIgnoreCase(MySQLConstants.PRIVILEGE_GRANT_OPTION_NAME)) {
+                    if (grant.isGrantOption()) {
+                        // WITH GRANT OPTION is enabled only in this case
+                        privilegeEnabled = true;
+                        break;
+                    }
```

Comment:
Excuse me, but is this change really needed since the condition is still the same?

Reply:
Nope, it is not the same.
Now, if we have `privilege.getName().equalsIgnoreCase(MySQLConstants.PRIVILEGE_GRANT_OPTION_NAME)`, we will go inside the block and avoid going to the condition `else if (grant.isAllPrivileges() || ArrayUtils.contains(grant.getPrivileges(), privilege))` for case `!grant.isGrantOption()`

Reply:
got it. Thanks